File: Aeds2/WarmUp/Prova 3/C [Warm-up 03] Desempilhando Caixas/main.c
#include <stdio.h>

int main()
{
    int quantPilhas = 0;
    int nCaixas = 0;
    int fim = 0;
    int aux = 0;

    scanf(" %d", &nCaixas);
    scanf(" %d", &quantPilhas);
    while (fim == 0)
    {
        int cont = 0;
        int achou = 0;
        int pos = 0;
        for (int i = 0; i < quantPilhas; i++) // olhando cada pilha
        {
            scanf(" %d", &aux); // numero de caixas na pilha
            for (int j = 0; j < aux; j++)
            { // olhando as pilhas
                int tmpCaixa = 0;
                scanf(" %d", &tmpCaixa);
                if (tmpCaixa == 1) // ao encontrar a caixa 1, guarda a posicao da caixa
                {
                    achou = 1;
                    pos = j;
                }
                else if (achou == 1 && j >= pos) // depois de encontrar a caixa 1 verifica se a posicao da outra caixa esta na mesma altura da caixa 1 ou acima
                // se estiver, joazinho vai ter que desempilhar +1 caixa;
                {
                    cont++;
                }
            }
        }
        printf("%d\n", cont);
        scanf(" %d", &nCaixas);
        scanf(" %d", &quantPilhas);
        if (nCaixas == 0 && quantPilhas == 0)
        {
            fim = 1;
        }
    }
    return 0;
}

File: Aeds2/WarmUp/Prova 3/C [Warm-up 03] Desempilhando Caixas/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String aux;
        int pos = 0;

        boolean fim = false;
        aux = scanner.nextLine();
        String[] partes = aux.split(" ");

        while (fim == false) {
            boolean achou = false;
            int desempilhado = 0;
            int quantCaixas = Integer.parseInt(partes[0]);
            int quantPilhas = Integer.parseInt(partes[1]);
            for (int i = 0; i < quantPilhas; i++) {
                aux = scanner.nextLine();
                partes = aux.split(" ");
                int numCaixasNaPilha = Integer.parseInt(partes[0]);
                int[] pilha = new int[numCaixasNaPilha];
                for (int j = 0; j < numCaixasNaPilha; j++) {
                    if (Integer.parseInt(partes[j + 1]) == 1) {
                        achou = true;
                        pos = j;
                    } else if (achou == true && j >= pos) {
                        desempilhado++;
                    }
                }
            }
            System.out.println(desempilhado);
            aux = scanner.nextLine();
            if (aux.compareTo("0 0") == 0)
                fim = true;
        }
        scanner.close();
    }
}
